apps/src/aichat/views/UserChatMessageEditor.tsx

```diff
@@ -16,14 +19,39 @@ const UserChatMessageEditor: React.FunctionComponent = () => {
     (state: {aichat: AichatState}) => state.aichat.isWaitingForChatResponse
   );
 
-  // TODO: If systemPrompt is undefined, handle this error case.
+  const userId: number = useSelector(
+    (state: {currentUser: CurrentUserState}) => state.currentUser.userId
+  );
+
+  const {currentLevelId, scriptId} = useSelector(
+    (state: {progress: ProgressState}) => state.progress
+  );
```

Comment:
nit: use `useAppSelector` here so you don't have to manually cast the `state` type

Reply:
Only for `aichat` state though - correct?

Reply:
`RootState` is the type of the global application state (for whatever slices have been ported to TypeScript already), so it will include `progress` as well

Reply:
Based on comment below, moving to `aichatRedux`. Thanks!